lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1702,6 +1702,7 @@ public function get_collection_params() {
 		$params['filter'] = array(
 			'description'       => __( 'Use WP Query arguments to modify the response; private query vars require appropriate authorization.' ),
 		);
+		$params['filter[posts_per_page]'] = $params['per_page'];
```

Comment:
HAHAHA, I know it looks like a cheat. :smile: 

But I found out some problem like following.

* Callback function for validation is called in the WordPress core. I couldn't figure it out that I should make ticket on WordPress Trac.
* `per_page` and `filter[posts_per_page]` are same parameter and `per_page` will over written by `filter[posts_per_page]`. The cause of this bug is the 'per_page' validation is being applied but the `filter[post_per_page]` validation is not being applied. They should both be using the same validation.

I have another idea.

I thought following is more readable?

lib/endpoints/class-wp-rest-posts-controller.php: 1705
```
$params['filter.posts_per_page'] = $params['per_page'];
```

plugin.php: 285
```
		// for validating filter args
		if ( isset( $request['filter'] ) ) {
			if ( $filter = preg_split( "/\./", $params ) && isset( $filter[1] ) && $filter[1] ) {
				$filters = $request->get_param( 'filter' );
				if ( ! empty( $filters[ $filter[1] ] ) ) {
					$value = $filters[ $filter[1] ];
				}
			}
		}
```